api/v1beta1/gcpmachine_types.go

```diff
@@ -252,6 +314,10 @@ type GCPMachineSpec struct {
 	// +kubebuilder:validation:Enum=Enabled;Disabled
 	// +optional
 	ConfidentialCompute *ConfidentialComputePolicy `json:"confidentialCompute,omitempty"`
+
+	// RootDiskEncryptionKey defines the KMS key to be used to encrypt the root disk.
+	// +optional
+	RootDiskEncryptionKey *CustomerEncryptionKey `json:"rootDiskEncryptionKey,omitempty"`
```

Comment:
Are you ever likely to expand a difference between this use of `CustomerEncryptionKey` and the other one? Typically we recommend not re-using structs as you can accidentally expand an API you didn't intend to, though in this case it may be safe

Reply:
In this case, the struct is used for disks only on (either AdditionalDisks or RootDisk) it seems we would want to keep the same encryption capabilities. The AttachedDiskSpec also has DiskType and size, the same as what's defined for RootDeviceType and RootDeviceSize.